### PR TITLE
fix(datasource): re-encode URL to support white spaces in Azure projects

### PR DESCRIPTION
**Pull request by lfcyja**
## Changes

The URL for GIT submodules are decoded as a side-effect of the `git-url-parse` node module. This leads to errors when using submodules with whitespaces in the URL, e.g. Azure DevOps project name.

Error: `"WARN: Error mapping git submodules during extraction"`, raised in `extract.ts` and originating from `index.js` in `git-url-parse`.

## Context

The `gitUrlParse` function respects the fact that in some hosting services such as Azure DevOps, you can use white-space characters in the repository and owner names. It URI-decodes the components accordingly and uses these to assemble the full_name property. This is done independently of the hosting environment.

When the `GitUrl` object is later used to compose the repository string, however, these components are not re-encoded. The full_name property is used to build the string.

I already tried to fix the issue in the affected submodule ([active PR here](https://github.com/IonicaBizau/git-url-parse/pull/156)), but the owner does not seem to maintain the repository anymore.
Hence, I attempted a fix in your module.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [X] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [X] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/git/url.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ export function getHttpUrl(url: string, token?: string): string {
     }
   }
 
-  return parsedUrl.toString(protocol);
+  return new URL(parsedUrl.toString(protocol)).href;
```
Comment (JamieMagee) [inline lib/util/git/url.ts:35]: Can you update or add a test to verify the new behaviour?

Comment (lfcyja) [reply lib/util/git/url.ts:35]: I added a new test to verify the extraction process.

**Review by JamieMagee — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Comment by viceice** [comment]
please stop merging main, GitHub merge queue will do it for you

**Comment by alexanderlinne** [comment]
@JamieMagee @viceice Hi, is there anything else to do to have this merged?

**Comment by rzsita** [comment]
Hi,
This issue doesn’t seem to be fixed as I’m still facing it with version 36.23.0.

**Pull request merged**